code/modules/projectiles/updated_projectiles/gun_helpers.dm

```diff
@@ -330,6 +330,10 @@ should be alright.
 		to_chat(user, "<span class='warning'>[attachment] doesn't fit on [src]!</span>")
 		return
 
+	if(overcharge == TRUE)
```

Comment:
How about something like this?
```
if(overcharge)
	toggle_chargemode(user)
```
That way you don't need to return here.
It would require for you to make `toggle_chargemode()` a gun proc that returns, redefined for the lasgun, though.